server/src/main/java/io/crate/analyze/PartitionPropertiesAnalyzer.java

```diff
@@ -57,7 +58,7 @@ public static PartitionName toPartitionName(RelationName relationName, List<Assi
 
         int idx = 0;
         for (Object o : properties.values()) {
-            values[idx++] = DataTypes.STRING.implicitCast(o);
+            values[idx++] = DataTypes.STRING.implicitCast(o, CoordinatorTxnCtx.systemTransactionContext().sessionSettings());
```

Comment:
Maybe move this outside of the loop to avoid creating a new context for each value.